strats.md

```diff
@@ -773,18 +938,120 @@ A `comeInWithTemporaryBlue` entrance condition indicates that Samus must come in
+- _remoteAndLandingMinTiles_: When entering a mockball, it takes some time to accelerate up to full speed, which means that even when using a remote runway (i.e. one not connected to the door) to gain speed for the jump, the amount of landing space in front of the door still matters. Depending on the strat, different combinations of remote runway and landing lengths may work (e.g. with shorter landing lengths possibly requiring longer remote runways to compensate). This property is a list of pairs, where in each pair the first value gives a minimal remote runway used to gain speed, and the second value gives the corresponding minimal amount of landing tiles in front of the door usable to gain speed at the start of the mockball. 
+
+A `comeInWithMockball` entrance condition must match with one of the following conditions on the other side of the door: `leaveWithMockball` or `leaveWithRunway`:
+
+- A match with `leaveWithMockball` has the following requirements:
+  - The `blue` property of `leaveWithMockball` must be "no" or "any".
+  - `remoteAndLandingMinTiles` must contain at least one pair `(minRemoteLength, minLandingLength)` such that the effective length of the `remoteRunway` is at least `minRemoteLength` and the effective length of the `landingRunway` is at least `minLandingLength`.
+  - The `canMockball` tech (including `Morph` item).
+- A match with `leaveWithRunway` has the following requirements:
+  - The effective runway length must be at least `acjacentMinTiles`.
+  - If the previous room is heated, then `heatFrames` are included based on the time spent running in that room. The minimally required heat frames are calculated the same way as in `comeInShinecharging`, except here there is no second runway to combine with. 
+  - If the previous door environment is water, then `Gravity` is required.
+  - The `canMockball` tech (including `Morph` item) is required.
+
+```json
+{
+  "name": "Come In With Mockball",
+  "notable": false,
+  "entranceCondtion": {
+    "comeInWithMockball": {
+      "adjacentMinTiles": 11,
+      "remoteAndLandingMinTiles": [[8, 2], [7, 3]]
+    }
+  },
+  "requires": []
+}
+```
+
+### Come In With Spring Ball Bounce
+
+A `comeInWithSpringBallBounce` entrance condition indicates that Samus must enter the room by spring ball bouncing in the doorway of the previous room. It has the following properties:
+
+- _adjacentMinTiles_: This is the minimum effective runway length in case an adjacent runway (connected to the door) is used to gain speed, jump, and bounce.
+- _remoteAndLandingMinTiles_: A list of pairs, where in each pair the first value gives a minimal remote runway used to gain speed, and the second value gives the corresponding minimal amount of landing tiles in front of the door usable for the bounce.
+- _movementType_: This takes one of three possible values, "mockball", "airball", and "any", indicating the type of bounce that is required.
```

Comment:
Airball is always used to mean Morphing while still travelling upward.  And when we talked about renaming `canLateralMidAirMorph` to `canAirball`, the problem is that it covers downwards morphs too.  There are enough downward morph techs now that maybe there is enough separation that LateralMidAirMorphs are only the rising ones, but it's also pretty likely that renaming it to airball is better even if there is technically a problem somewhere.

Reply:
Oh interesting. If I understand, there's no fundamental difference between doing it moving upward vs. moving downward, as far as the inputs pressed and the effect on the game physics; i.e. you end up in the same movement type/pose either way. So it's a little puzzling that the name is only reserved for moving upward. But I guess if that's how the community understands the word, then it is what it is. I can understand the upward movement version being thought of as different from a strat perspective, because of how it leaves you with uncontrollable upward speed, along with a reduced hitbox which could be used to delay bonking a ceiling; whereas the downward version would often involve a release and repress of jump to control the height. The exact same things are also true though of regular `canMidAirMorph` (from a stand) and we don't have separate names for upward vs. downward moving versions of that (though maybe we could?).

Would `controlled`/`uncontrolled` sound better, as a replacement for `mockball`/`airball`?

Reply:
sure